rust/arrow/src/array/equal/mod.rs

```diff
@@ -146,118 +146,103 @@ fn equal_values(
     rhs_start: usize,
     len: usize,
 ) -> bool {
-    // compute the nested buffer of the parent and child
-    // if the array has no parent, the child is computed with itself
-    #[allow(unused_assignments)]
-    let mut temp_lhs: Option<Buffer> = None;
-    #[allow(unused_assignments)]
-    let mut temp_rhs: Option<Buffer> = None;
-    let lhs_merged_nulls = match (lhs_nulls, lhs.null_buffer()) {
-        (None, None) => None,
-        (None, Some(c)) => Some(c),
-        (Some(p), None) => Some(p),
-        (Some(p), Some(c)) => {
-            let merged = (p & c).unwrap();
-            temp_lhs = Some(merged);
-            temp_lhs.as_ref()
-        }
-    };
```

Comment:
I liked this more. A bit more explicit what was happening here.

Reply:
Interestingly, I found an issue with this while fixing the failing integration tests. See my latest commit https://github.com/apache/arrow/pull/9093/commits/66bb98e240f9ca49a7abe2a00ce83a117a62b426.

I'll follow up with changes for your suggestions.

Reply:
Well, if it does not work it does not work and we scratch it ^_^